### DISCUSSION TITLE
Railroad Crossings with Light and Gates

### DISCUSSION THREAD

**Discussion by rankieth** [InterurbanDev/MinecraftRailroadBlocks#35] [Ideas]
It would be cool if we could make railroad crossings with lights, gates, and bells. It would also be cool if they were customizable, allowing us to edit the gate length (and what angle the gate is when it's up), number of light bulbs on a crossing (to allow for side lights and one sided crossing lights), and even what type of bell is used and whether or not the bell rings during the entire time the gate is down. It would even be cool if they could be compatible with create's trains. Thoughts?

![crossing](https://github.com/user-attachments/assets/7be4e045-904a-46c4-825e-3f4d4fddd003)
![crossing2](https://github.com/user-attachments/assets/3f84fbb5-f322-4a31-a6ac-d277a683d1a0)
![crossing3](https://github.com/user-attachments/assets/1c053f84-28b4-461e-ace2-9cd5f614d868)
![crossing4](https://github.com/user-attachments/assets/7b4c0f86-4925-47f5-8f0b-c95cb95140ad)

**Comment by InterurbanDev** [comment]
I certainly love the idea. I have tried working on getting some functional railroad gates and lights in the past, but I've had trouble figuring out how to implement it. Unfortunately, there's not a lot of documentation for how to do more advanced mod ideas, but it is definitely on the list of things to do.

That being said, if I do manage to get something working, it'll probably not be as advanced as this, but I'd love to see it if someone else has the skills and knowledge to do so. Thanks for the suggestion!
